server/api/controllers/class/index.js

```diff
@@ -2,46 +2,79 @@ const ModelsRoot = require('../../models');
 
 let Models = null;
 let db = null;
-let container = {};
+let classContainer = {};
 
-container.readOne = async (id) => {
+// TODO:PUSH:
+classContainer.readOne = async (id) => {
   try {
-    return await Models.class.findOne({
+    let record = await Models.class.findOne({
       where: { id: id }
     });
+
+    if (record instanceof Error) throw record;
+
+    return record;
   } catch (error) {
     return error;
   }
 };
 
-container.writeOne = async (data) => {
+// TODO:PUSH:
+classContainer.writeOne = async (data) => {
   try {
-    return await Models.class.create(data);
+    let record = await Models.class.create(data);
+
+    if (record instanceof Error) throw new Error('Record must be complete');
+
+    return record;
   } catch (error) {
     return error;
   }
 };
 
-container.updateOne = async (data) => {
+// TODO:PUSH:
+classContainer.updateOne = async (data, id, method) => {
   try {
-    return await Models.class.update(data, {
-      where: { id: data.id }
+    let record = await classContainer.readOne(id);
+
+    if (method === 'PUT') {
+      if (!record) {
```

Comment:
This creates a new record in a `PUT` request if the look up did not find a record to update.  This stays in compliance with [rfc2616](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.6)